plugin/src/main/java/com/playmonumenta/scriptedquests/utils/QuestUtils.java

```diff
@@ -4,33 +4,53 @@
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.Collections;
-import java.util.HashSet;
 import java.util.Set;
 
 import org.bukkit.ChatColor;
 import org.bukkit.command.CommandSender;
 import org.bukkit.plugin.Plugin;
+import org.checkerframework.checker.nullness.qual.Nullable;
 
 import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
 import com.google.gson.JsonObject;
 
 public class QuestUtils {
 	@FunctionalInterface
-	public interface JsonLoadAction {
+	public interface JsonLoadAction extends JsonLoadActionWithFile {
 		/**
 		 * Called with each file loaded into a JSON object
 		 * Return value is identifying string to return to the user or null
 		 */
 		String load(JsonObject object) throws Exception;
+
+		default String load(JsonObject object, File file) throws Exception {
+			return load(object);
+		}
+	}
+
+	@FunctionalInterface
+	public interface JsonLoadActionWithFile {
+		/**
+		 * Called with each file loaded into a JSON object
+		 * Return value is identifying string to return to the user or null
+		 */
+		String load(JsonObject object, File file) throws Exception;
 	}
 
 	public static void loadScriptedQuests(Plugin plugin, String folderName, CommandSender sender, JsonLoadAction action) {
-		Set<CommandSender> senders = new HashSet<CommandSender>();
-		senders.add(sender);
-		loadScriptedQuests(plugin, folderName, senders, action);
+		loadScriptedQuests(plugin, folderName, sender, (JsonLoadActionWithFile) action);
+	}
```

Comment:
I think this entire thing can be removed since JsonLoadAction inherits JsonLoadActionWithfile (the below block)

Reply:
These are necessary for lambda syntax to work (without a manual cast at least.)